pkg/commands/auth/actions/login.go

```diff
@@ -91,29 +98,23 @@ func Login() *cobra.Command {
 			}
 
 			eg, ctx := errgroup.WithContext(cmd.Context())
+			if personalToken == "" {
+				loginViaBrowser(eg, ready, cmdFlags, ctx)
+			}
 			eg.Go(func() error {
-				select {
-				case url := <-ready:
-					_, _ = fmt.Println("You will be redirected to your web browser to complete the login process")
-					_, _ = fmt.Println("If the page did not open automatically, open this URL manually:", url)
-					if err := browser.OpenURL(url); err != nil && cmdFlags.verbose {
-						log.Println("Could not open the browser:", err)
-					}
+				var oauthToken *oauth2.Token
+				var err error
 
-					return nil
-				case <-ctx.Done():
-					return fmt.Errorf("context done while waiting for authorization: %w", ctx.Err())
-				}
-			})
-			eg.Go(func() error {
-				token, err := oauth2cli.GetToken(ctx, cfg)
-				if err != nil {
-					return fmt.Errorf("could not get a token: %w", err)
+				if personalToken == "" {
+					oauthToken, err = oauth2cli.GetToken(ctx, cfg)
+					if err != nil {
+						return fmt.Errorf("could not get a oauthToken: %w", err)
+					}
 				}
 
-				session, err := storeProfileToken(cmdFlags.profileName, cmdFlags.clientID, apiURL, appURL, portalURL, token)
+				session, err := storeProfileToken(cmdFlags.profileName, cmdFlags.clientID, apiURL, appURL, portalURL, oauthToken, personalToken, ctx)
 				if err != nil {
-					return fmt.Errorf("could not store access token: %w", err)
+					return fmt.Errorf("could not store access oauthToken: %w", err)
 				}
```

Comment:
this go goroutine block is not necessary if you dont open the browser for login.